### PR TITLE
CI build for HelloWorld

### PR DESCRIPTION
**Pull request by rajdeepd**
**Related issues**
https://github.com/salesforce/TransmogrifAI/issues/114

**Describe the proposed solution**
Created a new circleci based configuration which checks for scalastyle and testcoverage

**Describe alternatives you've considered**
It will be easier for helloworld to have its own CI build.

**Additional context**
none

### CODE REVIEW

**Comment by rajdeepd** [comment]
@tovbinm is this a bug in the test case? https://travis-ci.com/salesforce/TransmogrifAI/jobs/171776690

**Comment by tovbinm** [comment]
@rajdeepd it's not running the helloworld build yet. The results seems to be from the regular build.

**Comment by rajdeepd** [comment]
@tovbinm is there something else I need to add or this will suffice.

**Comment by tovbinm** [comment]
@rajdeepd we have two options here:
1. configure helloworld as a separate build on Travis & Circle CI (this is what you started to do) by adding  build configs.
2. include helloworld build as part of the regular build (by modifying the build.gradle file in the root project). we can add a extra task that execute helloworld builds, which we will use on CI. E.g.:
```
./gradlew reportScoverage && ./gradlew publishToMavenLocal && ./gradlew buildHelloworld
```

File: helloworld/.circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+version: 2
```
Comment (tovbinm) [inline helloworld/.circleci/config.yml]: I dont think you need the `elloworld/.circleci/config.yml` file anymore

File: .circleci/config.yml [hunk 2]
```diff
@@ -46,6 +46,25 @@ jobs:
           name: Clone repo & compile project
```
Comment (tovbinm) [inline .circleci/config.yml]: `name: Compile project`

File: .circleci/config.yml [hunk 3]
```diff
@@ -46,6 +46,25 @@ jobs:
...
+            name: Scala style check for HelloWorld
+            command: cd helloworld && ./gradlew --parallel scalaStyle && cd ..
+  compile-hw:
+      <<: *machine-config
+      steps:
+        - checkout
+        - *open-jdk
+        - *restore-gradle-cache
+        - run:
+            name: Clone repo & compile project
```
Comment (tovbinm) [inline .circleci/config.yml]: `name: Compile Helloworld project`

File: .circleci/config.yml [hunk 4]
```diff
@@ -46,6 +46,25 @@ jobs:
...
+        - *open-jdk
+        - run:
+            name: Scala style check for HelloWorld
+            command: cd helloworld && ./gradlew --parallel scalaStyle && cd ..
+  compile-hw:
+      <<: *machine-config
+      steps:
+        - checkout
+        - *open-jdk
+        - *restore-gradle-cache
```
Comment (tovbinm) [inline .circleci/config.yml]: I dont think you need to `*restore-gradle-cache` for Helloworld.

File: .circleci/config.yml [hunk 5]
```diff
@@ -46,6 +46,25 @@ jobs:
...
+        - run:
+            name: Clone repo & compile project
+            command: cd helloworld && ./gradlew --parallel testScoverageClasses && cd ..
+        - *save-gradle-cache
```
Comment (tovbinm) [inline .circleci/config.yml]: also I dont think you want to save the cache for helloworld either

File: helloworld/.circleci/config.yml (hunk 1 above)
Comment (rajdeepd) [reply helloworld/.circleci/config.yml]: @tovbinm  i know that, will remove it

**Comment by rajdeepd** [comment]
> @rajdeepd we have two options here:
> 
> 1. configure helloworld as a separate build on Travis & Circle CI (this is what you started to do) by adding  build configs.
> 2. include helloworld build as part of the regular build (by modifying the build.gradle file in the root project). we can add a extra task that execute helloworld builds, which we will use on CI. E.g.:
> 
> ```
> ./gradlew reportScoverage && ./gradlew publishToMavenLocal && ./gradlew buildHelloworld
> ```
> I think the second option is preferred, cause you would be able to test it locally and we wont need to replicate the CI build configs into helloworld project.
> 
> Does it make sense?

@tovbinm yes

File: .circleci/config.yml (hunk 5 above)
Comment (rajdeepd) [reply .circleci/config.yml]: fixed

File: .circleci/config.yml (hunk 4 above)
Comment (rajdeepd) [reply .circleci/config.yml]: fixed

File: .circleci/config.yml (hunk 3 above)
Comment (rajdeepd) [reply .circleci/config.yml]: fixed

File: .circleci/config.yml (hunk 2 above)
Comment (rajdeepd) [reply .circleci/config.yml]: fixed

**Comment by rajdeepd** [comment]
@tovbinm sure!

**Comment by tovbinm** [comment]
Thank you @rajdeepd Looking forward to it ;)

**Pull request merged**